content/ko/distributed-systems.md

```diff
@@ -0,0 +1,32 @@
+---
+title: 분산 시스템 (Distributed System)
+status: Completed
+category: 개념
+tags: ["아키텍처", "", ""]
+---
+
+## 개념
+
+분산 시스템이란 네트워크를 통해 연결된 자율 컴퓨팅 요소들의 모음으로, 
+사용자에게 하나의 일관성있는 시스템으로 보여진다. 
+일반적으로 [노드(nodes)](/nodes/)라고 언급되는 구성 요소는
+하드웨어 장치(예: 컴퓨터, 휴대폰) 또는 소프트웨어 프로세스일 수 있다.
+노드들을 공통의 목표 달성을 위해 프로그래밍 하고, 협업하기 위해 네트워크를 통하여 메시지를 교환한다.
+
+## 다루는 문제
+
+오늘날의 수많은 최신 애플리케이션은 너무 커서 작동하려면 슈퍼컴퓨터가 필요하다.
+Gmail 또는 Netflix를 생각해보면, 전체 애플리케이션을 호스팅(host)할 만큼 충분한 성능을 가진 단일 컴퓨터는 존재하지 않는다.
+여러 대의 컴퓨터를 연결하면 컴퓨팅 성능은 거의 무제한이 된다.
+분산 컴퓨팅 없이는 오늘날 우리가 의존하는 대부분의 어플리케이션이 있을 수 없었을 것이다.
+
+전통적으로, 개별 시스템은 수직적으로 [확장(scale)](/scalability/)한다. 즉, CPU나 메모리를 더 추가할 때를 나타낸다.
+수직적 확장은 시간이 많이 걸리고, 중단 시간을 필요로하며, 한계에 빠르게 도달한다.
+
+## 문제 해결 방식
+
+분산 시스템에서는 [수평적 확장(horizontal scaling)](/horizontal-scaling/)이 가능하다(예: 필요시 시스템에 노드 추가).
+수평적 확장을 자동화할 수 있어 시스템이 갑작스러운 작업량(workload) 또는 리소스 소비 증가를 처리할 수 있게 만든다.
+
+분산되지 않은 시스템은 고장 위험에 노출되는데, 이는 하나의 시스템에 장애가 발생하면 전체 시스템에 장애가 발생하기 때문이다.
```

Comment:
```suggestion
분산되지 않은 시스템은 장애 위험에 노출되는데, 이는 단일 머신(machine)의 장애가 전체 시스템에 장애로 이어지기 때문이다.
```
원문을 고려하여 수정해보았습니다.

A non-distributed system exposes itself to risks of failure because if one machine fails, the entire system fails.